### ISSUE TITLE
DataTable Checkbox: Accessibility - ARIA attributes should be valid for the element and ARIA role to which they are assigned

### ISSUE THREAD

**Issue by Aryan3902** [primefaces/primereact#8227]
### Describe the bug

Running IBM Accessibility Checker on any DataTable with Checkbox based select functionality gives the following violation:

The ARIA attributes "aria-checked" are not valid for the element <input> with implicit ARIA role "checkbox"

Element location

```
<input
    data-pc-section="input"
    aria-label="All items selected"
    aria-checked="true"
    aria-invalid="false"
    tabindex="0"
    class="p-checkbox-input"
    type="checkbox">
```

What to do
Remove ARIA state or property attributes that are not valid for the element roleOr, change to a role that supports the intended behavior of the element, using the ARIA specification as a guide

In the following example, the element references only valid ARIA attributes for an element with the role of "tree", a global aria-labelledby property, and an aria-expanded state:

```
<ul id="tree1" class="tree root-level" role="tree" aria-labelledby="label_1">
    
<li id="fruits" class="tree-parent" role="treeitem" tabindex="0" aria-expanded="true">
    <span>Fruits</span>
```

Rule ID: aria_attribute_valid
Reason ID: Fail_invalid_implicit_role_attr

<img width="1915" height="1085" alt="Image" src="https://github.com/user-attachments/assets/586ee4bd-2fc7-412e-9747-9e443a313743" />

### System Information

```Shell
Can be reproduced in every environment using IBM Accessibility Checker
```

### Steps to reproduce the behavior

1. Create a DataTable component with selectionMode as checkbox
2. Run IBM Accessibility Checker Scan
3. Check violations
